src/config.ts

```diff
@@ -0,0 +1,129 @@
+import deepmerge from 'deepmerge';
+import { typeSymbol } from '@map-colonies/schemas/build/schemas/symbol';
+import configPkg from 'config';
+import semver from 'semver';
+import lodash, { type GetFieldType } from 'lodash';
+import { getEnvValues } from './env';
+import { BaseOptions, ConfigOptions, ConfigInstance } from './types';
+import { loadSchema } from './schemas';
+import { getOptions, initializeOptions } from './options';
+import { getRemoteConfig, getServerCapabilities } from './httpClient';
+import { ajvConfigValidator, validate } from './validator';
+import { createDebug } from './utils/debug';
+import { LOCAL_SCHEMAS_PACKAGE_VERSION } from './constants';
+import { createConfigError } from './errors';
+
+const debug = createDebug('config');
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+const arrayMerge: deepmerge.Options['arrayMerge'] = (destinationArray, sourceArray) => sourceArray;
+
+/**
+ * Retrieves the configuration based on the provided options.
+ *
+ * @template T - The type of the configuration schema.
+ * @param {ConfigOptions<T>} options - The options for retrieving the configuration.
+ * @returns {Promise<ConfigInstance<T>>} - A promise that resolves to the configuration object.
+ */
+export async function config<T extends { [typeSymbol]: unknown; $id?: string }>(
```

Comment:
$id is always defined
```suggestion
export async function config<T extends { [typeSymbol]: unknown; $id: string }>(
```